app/controllers/movies_controller.rb

```diff
@@ -1,17 +1,19 @@
 class MoviesController < ApplicationController
   def index
-    @movies = Movie.all.order(title: :asc).paginate(page: params[:page], per_page: 20)
+    attribute = params["sort_by"] || :title
+    order = params["sort_order"] || :asc
+    @movies = Movie.all.order(attribute => order).paginate(page: params[:page], per_page: 20)
```

Comment:
Instead of querying the database for all movies and the filtering, we should apply the filtering  directly when that parameter is present.
Imagine your database has 10,000 movies but only 1 called `Harry Potter and the Goblet of Fire`. Then you would first obtain all 100,000 movies to then select just the 1 you're interested in.